lib/rules/at-rule-prelude-no-invalid/index.mjs

```diff
@@ -0,0 +1,75 @@
+import { lexer } from 'css-tree';
+
+import { isRegExp, isString } from '../../utils/validateTypes.mjs';
+import isStandardSyntaxAtRule from '../../utils/isStandardSyntaxAtRule.mjs';
+
+import optionsMatches from '../../utils/optionsMatches.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'at-rule-prelude-no-invalid';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule, prelude) => `Unexpected invalid prelude "${prelude}" for at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-prelude-no-invalid',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary, secondaryOptions) => {
+	return (root, result) => {
+		const validOptions = validateOptions(
+			result,
+			ruleName,
+			{ actual: primary },
+			{
+				actual: secondaryOptions,
+				possible: {
+					ignoreAtRules: [isString, isRegExp],
+				},
+				optional: true,
+			},
+		);
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules((atRule) => {
+			if (!isStandardSyntaxAtRule(atRule)) return;
+
+			if (optionsMatches(secondaryOptions, 'ignoreAtRules', atRule.name)) return;
+
+			const { name, params } = atRule;
+
+			const { error } = lexer.matchAtrulePrelude(name, params);
+
+			if (!error) return;
+
+			if (error.name !== 'SyntaxMatchError') return;
+
+			const atName = `@${name}`;
+
+			const index = atName.length + 1;
```

Comment:
[suggestion] Using the `atRuleParamIndex` utility seems more accurate:

https://github.com/stylelint/stylelint/blob/a0c6dc43d32f3c93e7bcef090c06aefd59373e08/lib/utils/nodeFieldIndices.mjs#L12